% \externaldocument{3/chapter3.tex}
\section{Biological Regulation}
\label{sec:bioreg}
Information, \ie negative entropy, is present throughout the cosmos at every order of magnitude, and as its inhabitants, informatavores are witness among many of information's scales. The most fundamental operating units of information processing yet uncovered in biological systems are the RNA and DNA molecules. Reading the molecules which string together to form chains, these mega-molecules confer blueprints to the functional units of life, proteins. The mechanisms by which these instructions are read, \eg their order, duration, frequency \etc, are a major component of modern biological investigation. Together, these procedural tendencies have developed under many \emph{Goldilocks} conditions in a random manner, perhaps directed only through the goal of perpetual negative entropy degradation \citep{england2013statistical}. To tackle this goal with the highest likelihood of success, these regulatory relationships developed to enable response to any existing stimuli yet encountered. Less frequently and by luck some individuals may deviated to account for stimuli which had not yet come to pass but which could. Like evaluating and recalculating for each hand at a blackjack table in order of their probability in accordance to cards already at play, the ability to continue processing negative entropy is ensured by nondeterministically betting on what has been, what is and what could be. Such is the ``safety in numbers'' paradigm, wherein chances of individual survival are overlooked in the context of group persistence, which is a good trade-off for the universe under this hypothesis of its want to completely degrade negative entropy.

Generally speaking, evolution by way of natural selection is the major, nondeterministic mechanism of perpetuating lifeforms. Once initiated, this process bound only by biological principles within the constraints of fundamental forces, seems only to require time to breath forth its continuum of deviating forms. Building functional relationship among these ``endless forms most wonderful and most beautiful''\citep{darwin1869origin}, natural evolution has birthed a level of information storage and processing in the form of living organisms. In the right setting, the DNA molecule allows for not only the high fidelity storage of any set of biological blueprints (partially through its own replication), but the generation of various forms of more pliable, actionable RNA molecules to carry out its will\citep{roeder1991complexities}. In unison with post-translational factors, transporters, \etc, these various RNA molecules dictate the expression of proteins which carry out the life process, including all mechanisms of reading and repairing progenitor molecules. Regulation of influence whereby one or more molecules (directly or indirectly) dictate the behavior of another surely developed concurrently with oversight of that regulation by other molecules. This growing complexity could only have survived the harsh environmental changes brought on by each epoch and eon through more reliable and expansive information storage and retrieval, allowing for more creative and elaborate solutions to most natural environment yet exposed to life \eg growth via novel carbohydrate ``-oses''. Looking at such complexity without the aid of the very time it took to develop each sequential development has only recently become a tractable problem interpretable through the modern hominid alliance with outboard processors.

Science has long sought to understand the regulatory mechanisms guiding living bodies. Past attempts have placed constraints on possible regulatory capabilities, however many have since been overturn or amended through further, quasi-\emph{post-modern} study (famously the popular interpretation of Crick's central dogma \citep{crick1958protein}). In the light of such overturned or amended dogma, any similarly bold claim seems na\"{i}ve in the extreme. Today it is understood that bodies are composed of proteins that act in concert with not only those constituents, but with its surrounding environment. Crucially, proteins have come upon a means to affect change in other proteins behavior largely through binding, \eg regulating expression or enzymatic activity, which can be used to develop the array of living functions, such as building ion gradients along a protein-lipid membrane, (indirectly) feeding the engine to build said membrane. Regulation of these highly flexible, redundant and robust systems comes down to cause and effect, \ie one player dictating the behavior of another. Do not think this discourse ends so cleanly though, for often when mapped these regulations are hairball-like \citep{schulz2013grooming}, links seemingly departing from each node while also arriving at every other node (\ie a clique, see \cref{sec:pat} an unlikely biological prospect). In the specific domain of gene regulation explored here, this entails the regulation of a gene's transcription by any flavor of aptly named transcription factor (TF). This regulation varies in how much any given ``coding" gene's product is expressed by the various cellular machinery as an RNA molecule on its way to be separately regulated, processed and translated into a protein. Whether investigating proliferation of cell subtypes within a heterogeneous cancer mass \citep{weinberg1996cancer} or the tumbling quorum sensing of bacterial flagella used to seek out environments advantageous to its energy cycle, reliable roads of communication among machinery are crucial to continued survival.

Characterizing relationships is possible by measuring gene expression, dictated at the level of transcription through the physical binding of a TF upstream from a gene's start codon (any nucleotide (alphabet of DNA) triplet, \cref{fig:DNA}), likely to some degree to then be translated and processed to actionable protein. These TF binding relationships are often highly specific, but because of robust functionality can also be somewhat promiscuous, regulating multiple genes whose products act in unison toward some complimentary function or completely independent (TFs also useful in building networks or acting as priors, see \cref{sec:bipart}, \citep{ogris2017funcoup}). Since TFs are nothing more than proteins themselves, the protein product of any regulated gene can then itself play a regulatory role on any number of other genes (notice the overall cyclic nature of \cref{fig:DNA}). Hence the problem evolves from one of strictly direct relations to include those secondary, \etc indirect relations, begetting the hairball of interactions which together governs the life process to some extent. In any given condition the life form may rely on any path from one node to the next to accomplish its objective, \eg tumbling towards increased sucrose gradient \citep{di2005chemogenomic}. The question may arise to the fitness advantage of such redundant ``pathways'' to carry out similar functionality \citep{layek2011biological}. This is where one must not only consider the vast combinatorial power of nature to stimulate, but also the speed to functionality over time. Delayed onset of function can prepare secondary responses to environmental assaults, among other capabilities, and so while the end goal of pathways may look identical in hindsight, details of the cell's present may necessitate the subtle variation in when the response is brought into play (see \cref{sec:regwhatwhen}). So crucial are these relationships to the wild-type functioning of any cell, however, as is to be expected, when operating over many independent variables, direct relationships are often muddling and ambiguous. Creating ordered, predictable responses to internal and external stimuli is at the heart of the life process, and these are precisely what we mean to uncover.

%%mention next gen seq: ion/454-pyro (SU)/sanger/illumina/solid
Several methods are available for quantifying and characterizing biology to allow resolution of regulatory machinery through perturbation-based gene regulatory network (GRN) inference, and generally include but are not limited to microarray, qPCR, RNA-Seq, transcriptome methods and finally survival/phenotypic assay \citep{ideker1999discovery,bansal2007infer,castelo2009reverse,cosgrove2008predicting}. As with all developing technologies there are trade-offs in what specifically is under investigation, and what can be overlooked to afford that focus. As may be expected, these various techniques are positioned with advantage to certain levels of biology and their collective characterizations are thus segregated and gathered in (often isolated) databases. Known interactions are very similarly cataloged, where known TF binding interactions in the model organism \emph{E.coli} are housed in \emph{RegulonDB}\citep{gama2008regulondb}, while \emph{Yeastract}\citep{teixeira2006yeastract} houses those for \emph{S.cerevisiea}. These are quite useful when gold standards are sought to check against networks inferred from new platforms, to validate findings and classify possible novel interactions \eg say as stemming from known hub genes, \etc.

\begin{figure}%[H]
\centering
\includegraphics[width=1\linewidth]{2/DNA2.eps}
\caption{\textbf{Agnostic Biological Regulation.} Elements are seen to regulate one another regardless of biological mode/level, leading to loops as well as cascades of regulatory signaling. The regulation need not be direct, as this example contains no two factors directly linked but rather factors are linked between both the three distinct biological levels of DNA Sequence (S), RNA Gene (G) and Protein (P) as well as within level, \ie $G2\to G1$, $P3\to P2\to P1$. In this example, S1 up-regulates G1, which itself is also up-regulated by G2, which concomitantly down-regulates G3 and P3. P3 would normally directly bind both P1 and P2, however the partial absence of both now limit up-regulating expression of S2 to initiate further regulation (not shown).
}
\label{fig:DNA}
\end{figure}


\subsection{Implication}
\label{sec:practical}
A better understanding of the interplay between constituent regulators throughout every level of biology is crucial toward the development of any form of personalized or precision medicine \citep{barabasi2011network}. A recent study linked a large portion of human genes, some 17,000 to over 15,000 diseases, disorders or abnormal phenotypes \citep{pinero2015disgenet}, making for nearly half a million unique gene-disease associations. Only by placing these components together can we glean practical insights towards actionable intervention in the general systemic decay leading to disease onset, development and progression \citep{flores2013p4}. Genotyping of certain disease markers simplifies the search-space in individual patients, making it possible to give reasonable developmental predictions, many of which have actionable responses to prevent any (further) future damage to the patient, \eg BRCA1 \citep{lerman1996brca1}. Such relationships are continually uncovered, revealing more of nature's order, enabling, for example, ever wider fetal defect screening for expectant mothers, such as heart defects \citep{hyett1999using} as well as for those in especially compromising climates around the world vulnerable to certain debilitating disease.

In addition to identifying biomarkers, such regulatory networks can draw upon knowledge describing disease in other contexts, \eg clustering SNP based network motifs to complex traits across tissues via quantitative trait loci (QTL) \citep{platig2016bipartite}. This is done under a similar leading hypothesis when calculating orthologous genes across species, \ie same genes same function, although this is not always the case as environment is key. Finding similarities lets one then deduce tissue specific features stemming from novel motifs. Thus, while SNPs have classically been a resource bloating databases but seldom used to describe systems, in this way regulatory networks can enrich databases with new knowledge, opening SNP and other such isolated data to use by the community. Furthermore, new applications could developed, \eg more simply profiling new patients, expanding and repurposing the intended use of such an existing resource. 

Drug development is another area of huge potential upon accurately mapping the interaction among biomolecules. Understanding the relationships present in a individual and tracking them as they change during disease development allows for one to uncover exact targets for drug intervention \citep{schreiber2000target}. It is not so simple to say once a target is known a treatment can be found, but it is has long been a necessary precursor of such development. What is more promising still is the targeting of gene modules, complexes or regulatory subnetworks so interdependent that knocking out one element alters the activity of the rest. Such targeting would offer a nevertheless real means of affecting change in patients without the extreme specificity required of singular targets. A growing field lies in the use of a constrained GRN topology, directed acyclic graphs (DAGs, see\cref{sec:pat}), among others, to screen profiles of compounds, to repurpose adverse side effects to affect change in disparate classes of patients \citep{alaimo2016recommendation}. The more completely this interactive landscape is detailed, the more directly compounds targeting similar combinations of sites in antagonism with disease progression can be identified, not to mention the prospect of designing generic molecules. This lessened specificity would, however, bring concern for effects beyond design. Whether by repurposing, or repositioning, drugs tested to be safe for human consumption, or designing new drugs to treat new disease, accurate knowledge of the mechanisms underlying developmental paths toward disease are key, an aspect of which can be gleaned from reliable GRN inference.
